### ISSUE TITLE
🐞 Editor lags while resizing

### ISSUE THREAD

**Issue by Wouter01** [CodeEditApp/CodeEditTextView#119]
### Description

When resizing the editor, the whole app will lag quite a bit.
I've had a look and the culprit seems to be the rulerview:
```swift
rulerView = STLineNumberRulerView(textView: textView, scrollView: scrollView)
```

When this is commented out, everything is (mostly) smooth.

Another thing I noticed is that if `textView.widthTracksTextView` is set to false, it's also smoother.

### To Reproduce

1. Open editor with file
2. Resize the window

### Expected behavior

Resize is smooth

### Version information

CodeEdit: [e.g. 1.0]
macOS: [e.g. 13.1]
Xcode: [e.g. 14.2]

**Comment by thecoolwinter** [comment]
After some investigation, we’ll need to roll our own ruler view that doesn’t re-draw every line every time it’s invalidated. We should be able to only draw visible line numbers. That will help with the lag the ruler view is causing. 

We still need to look into the lag being caused by the text view itself further.

**Comment by thecoolwinter** [comment]
Fixed with #138
